src/apicov.py

```diff
@@ -27,6 +31,37 @@ def upload_coverage_data(coverage_data, api_key):
         return False
 
 
+def create_gcov_archive(coverage_instance, output_path=None, archive_name="gcov_files.zip"):
```

Comment:
```suggestion
def create_gcov_archive(coverage_instance: LibCoverage, output_path: str | None=None, archive_name: str | None="gcov_files.zip") -> str | None:
```

Although all this info is in the docstring, it's a lot clearer to include it as type annotations, too. That also opens up the possibility of enabling type checking with `ty` or `mypy` down the line.